stubs/python-jose/jose/backends/base.pyi

```diff
@@ -1,17 +1,23 @@
+from _typeshed import Self
 from typing import Any
 
 class Key:
+    # Enable when we can use stubs from installed dependencies,
+    # as `key` can be of type cryptography.x509.base.Certificate:
+    # from cryptography.x509 import Certificate
```

Comment:
I am not sure most of the values will be available, and adding the types as a dependency seems a bit of an overkill, also knowing they are outdated.

In any case, things like `cryptography_backend` don't exist in the current stubs, I think it should be typed `cryptography.hazmat.backends.openssl.backend.Backend` but that does not seem to exist in the current stubs. So I was thinking it might be better to leave all as they are and just type them whenever it's possible with actual values 😕 

If it is useful, maybe I could go over a few of them and try to type them if they don't depend on `cryptography`, as I did for other backends.

Reply:
> If it is useful, maybe I could go over a few of them and try to type them if they don't depend on cryptography, as I did for other backends.

I've never used these stubs or `python-jose` itself, so I can't comment on whether it's worth your time to do that :) Feel free to skip it if you don't think it would be useful 👍

Reply:
It's the first time for me as well, I'm adding stubs as I need them and appreciate a lot the time y'all take to review them 🙏  If that's ok with you, I think I'll leave it like this for now, will further improve in a separate PR if I find some more headaches 🙂